AdvLoggerPkg/Library/AdvancedLoggerLib/DxeCore/AdvancedLoggerLib.c

```diff
@@ -400,12 +403,24 @@ DxeCoreAdvancedLoggerLibConstructor (
   if (LoggerInfo != NULL) {
     mAdvLoggerProtocol.LoggerInfo = LoggerInfo;
     mLoggerInfo->TimerFrequency   = GetPerformanceCounterProperties (NULL, NULL);
-    Status                        = SystemTable->BootServices->InstallProtocolInterface (
-                                                                 &ImageHandle,
-                                                                 &gAdvancedLoggerProtocolGuid,
-                                                                 EFI_NATIVE_INTERFACE,
-                                                                 &mAdvLoggerProtocol.AdvLoggerProtocol
-                                                                 );
+
+    PreDxeLogsHobEntry = GetFirstGuidHob (&gAdvancedLoggerPreDxeLogsGuid);
+
+    if (PreDxeLogsHobEntry != NULL) {
+      PreDxeLogs = *(ADVANCED_LOGGER_PRE_DXE_LOGS_HOB *)GET_GUID_HOB_DATA (PreDxeLogsHobEntry);
+      if (PreDxeLogs.Signature != ADVANCED_LOGGER_PRE_DXE_LOGS_SIGNATURE) {
+        ASSERT (PreDxeLogs.Signature == ADVANCED_LOGGER_PRE_DXE_LOGS_SIGNATURE);
+      } else {
+        AdvancedLoggerMemoryLoggerWrite (DEBUG_INFO, (CONST CHAR8 *)(UINTN)PreDxeLogs.BaseAddress, PreDxeLogs.LengthInBytes);

```

Comment:
As discussed offline, this write call will save all the content into advanced logger region as a single adv logger entry. Might want to make the output from adv logger parser works as expected.